icu4c/source/common/cjmlbe.h

```diff
@@ -0,0 +1,170 @@
+// © 2022 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#ifndef CJMLBREAKENGINE_H
+#define CJMLBREAKENGINE_H
+
+#include "unicode/utypes.h"
+#include "unicode/uniset.h"
+#include "unicode/utext.h"
+
+#include "brkeng.h"
+#include "dictbe.h"
+#include "hash.h"
+#include "uvectr32.h"
+
+#include <queue>
+#include <thread>
+#include <functional>
+#include <mutex>
+#include <condition_variable>
+
+
+U_NAMESPACE_BEGIN
+
+#if !UCONFIG_NO_NORMALIZATION
+
+/**
+ * A thread pool used to store a thread set to evaluate the feature in parallel.
+ */
+class ThreadPool {
+public:
+    ThreadPool();
+    ~ThreadPool();
+    void start();
+    /**
+     * Assign the task to a thread in the pool.
+     *
+     * @param task A task to be executed.
+     */
+    void addTask(const std::function<void()>& task);
+    /**
+     * Stop all the thread executions.
+     */
+    void stop();
+    /**
+     * Wait for all the threads in the pool to be finished.
+     */
+    void waitFinished();
+
+private:
+    void loop();
+    const int32_t poolSize = 3;
+
+    bool should_terminate = false;           // Tells threads to stop looking for jobs
+    std::mutex queue_mutex;                  // Prevents data races to the job queue
+    std::condition_variable mutex_job; // Allows threads to wait on new jobs or termination
+    std::vector<std::thread> threads;
+    std::queue<std::function<void()>> jobs;
+    std::condition_variable mutex_finish;
+    int32_t busy;
+};
+
+
+/**
+ * A machine learning break engine for the phrase breaking in Chinese and Japanese.
+ */
+class CjMLBreakEngine : public UMemory {
+
+public:
+    /**
+     * <p>Constructor.</p>
```

Comment:
No need for `<p>` in doxygen for C++. Empty lines create paragraph breaks.

Reply:
Done. Remove it.